### ISSUE TITLE
Random characters keep being repeated and require ibus restart

### ISSUE THREAD

**Issue by falzard** [ibus/ibus#1918]
This bug occurs on ubuntu 14.04

When the computer or the application freezes, then whatever you were typing will be randomly repeated after it unfreezes. So if you are freezing while typing "hello" in geany, then if you keep pressing any letter, it will randomly add "llo".
If you press "eeeeeeeeeeeeeeeeeeeeee" it will transform into "eeeeeeelloeeeeeeeeeelloeeeeeeeello". If you press backspace or ctrl too.

If you do this in firefox, it will happen only in firefox (happens to me very often). So it depends on the application.
Then the only way to solve this is to kill ibus or restart firefox.

Another downside of that horrible bug, is that it is completely bugging the program AutoKey.

It is described here as well:
http://askubuntu.com/questions/526066/random-character-input-starts-after-typing-during-short-freezes
https://bugs.launchpad.net/ubuntu/+source/lubuntu-meta/+bug/1325987
https://ubuntuforums.org/showthread.php?t=2230701

PS: It also occurs that in Nautilus you can't type anything, I don't know if it's related

**Comment by fujiwarat** [comment]
The random strings are output of the buffered strings.
Yes, it would be a bug in ibus but I think the random characters does not happen if the freezing problem is fixed and the freezing problem itself is not caused by ibus.
The DBus traffic might be crowded by someone and it affects applications which uses DBus likes IBus.
I'd suggest to fix the root cause instead of the IBus random chars since it hard for me to reproduce the freezing problem.

**Comment by falzard** [comment]
The computer can freeze for a ton of reasons. Sometimes you open an overloaded web page, sometimes the desktop freezes for whatever reason etc...

If I understand right, you are suggesting to fix all bugs that can cause freezing?

**Comment by fujiwarat** [comment]
> The computer can freeze for a ton of reasons. Sometimes you open an overloaded web page, sometimes the desktop freezes for whatever reason etc...

No, I cannot reproduce your issues. Try to describe step by step.

**Comment by falzard** [comment]
What?

If your computer freezes but it's NOT related to ibus, then ibus is affected and I refer you to my first post then ;)
http://askubuntu.com/questions/526066/random-character-input-starts-after-typing-during-short-freezes etc...

**Comment by fujiwarat** [comment]
Again, I cannot reproduce any freeze problems in either Ubuntu 16.04.2 and Fedora.
Try to describe step by step.

Seems your problem is Ubuntu.

**Comment by falzard** [comment]
The freeze is not related to ibus, it's sometiems a bug in ubuntu or firefox or whatever. And after the freeze is gone (because that's how it works usually, your computer is slowed down or irresponsive and then it works again), then the bug is here. It's described in each and every link I put above!

**Comment by fujiwarat** [comment]
> It's described in each and every link I put above!

I think it's not enough. Probably some conditions are missed to reproduce your problem. That's why I asked to describe step by step but not just adding links.
Also your problem would not happen in Fedora and I think it's not an upstream issue.
I can help to fix Ubuntu issues but I think it's not mandatory for me and should be fixed in the distribution side.
If I can always reproduce your problem, I will try to fix your problem and root cause.

> it's sometiems a bug in ubuntu or firefox or whatever.

I mean your problem would be caused by a special configuration in Ubuntu but I agree we are not clear where is the root cause.
The best way is that you investigate which difference causes your problem between Fedora and Ubuntu and then probably I can reproduce your problem with a modified Fedora.

**Comment by falzard** [comment]
Ok, but you can reproduce a computer overload on demand?

When a freeze will happen I'll tell you how it happened then lol, I don't know how to do otherwise.

**Comment by fujiwarat** [comment]
> Ok, but you can reproduce a computer overload on demand?

I looked at the buffer outputs the unnecessary characters about ten times but the next typings were always normal without any problems so I have no reproducing steps and I have not thought the problem is critical since only one typing caused the problem by each case.
If I cannot reproduce the problem, I cannot debug the it. That's why I asked you the reproducing steps.

**Comment by falzard** [comment]
I cannot reproduce it on demands neither, it's just when my computer is overloaded.

**Comment by tinyplasticgreyknight** [comment]
You can use the `stress` program to simulate heavy system load on demand, please give it a try.

**Comment by fujiwarat** [comment]
I tried `stress --cpu 1 --io 4 --vm 2 --vm-bytes 500M --timeout 10s` 30 times but don't see any problems.
